internal/services/pubkey_service.go

```diff
@@ -99,6 +117,48 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkeyResource, err := dao.GetPubkeyResourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	resources, err := pubkeyResource.ListByPubkeyId(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	for _, res := range resources {
+		if res.Handle != "" {
+			arn, err := sourcesClient.GetArn(r.Context(), res.SourceID)
```

Comment:
I believe this would fix the issue:
```suggestion
			var arn string
			arn, err = sourcesClient.GetArn(r.Context(), res.SourceID)
```

But I'd love to see a better solution :) @lzap ? :stuck_out_tongue:

Reply:
Well, the alternative is to use a different name, like `errArn` or `arnErr`. That's what I tend to use.

We do not like shadowing of variables because there can be some nasty bugs sometimes.